flopy/discretization/modeltime.py

```diff
@@ -7,61 +12,211 @@ class ModelTime:
 
     Parameters
     ----------
-    stress_periods : pandas dataframe
-        headings are: perlen, nstp, tsmult
-    temporal_reference : TemporalReference
-        contains start time and time units information
+    perlen : list, np.ndarray
+        list or numpy array of stress-period lengths
+    nstp : list, np.ndarray
+        list or numpy array of number of time-steps per stress period
+    tsmult : list, np.ndarray
+        list or numpy array of timestep mult infomation
+    time_units : int or str
+        string or pre-mf6 integer representation (ITMUNI) of time units
+    start_datetime : various objects
+        user supplied datetime representation. Please see the
+        ModelTime.datetime_from_user_input documentation for a list
+        of the supported representation types
+    steady_state : list, np.ndarray
+        optional list or numpy array of boolean flags that determine identify
+        steady-state or transient stress-periods
     """
 
     def __init__(
         self,
-        period_data=None,
```

Comment:
I know period data will be computed dynamically from perlen/nstp/tsmult now, but maybe still convenient to pass in a dataframe/recarray/list of tuples.. would it be hard to keep support for that?

Reply:
It's possible to keep support for that, but if we added that too it would create multiple ways to instantiate the class. I personally think we should have a single standard way (either `period_data` or by each variable that would be in a period_data array). But I'm open to having multiple ways, it just complicates the code

Reply:
I'd personally vote to keep the initializer parameter a table, for backwards compatibility and because it is consistent with what we do for packages. We could have a `from_arrays(perlen, nstp, tsmult)` class method?

In future we could maybe also consider e.g. a `from_times(times)` class method doing what [imod-python does](https://deltares.github.io/imod-python/user-guide/07-time-discretization.html#basics)